core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -91,6 +106,8 @@
     @Nullable
     private SearchShardTarget shard;
 
+    private transient Text index;
```

Comment:
The problem here is that the response often only contains the "_index" field (other information necessary to build a complete SearchShardTarget are only rendered out under certain conditions, e.g. if explanations are set). So in order to parse the "_index" name back, I either need a place to store it independently of a complete SearchShardTarget, or I need to construct a dummy SearchShardTarget with bogus values. This is the first option, I think its better than to construct a SearchShardTarget full of dummy values.